### PR TITLE
JP-3087: Fix zeroing of NaNs in straylight

### PR DESCRIPTION
**Pull request by drlaw1558**
This PR fixes a new issue where the MRS straylight routine converts NaNs to zeros.  Previously this was a safety catch in case any NaNs were present that were not meant to be there, but now that the rate file is meant to contain NaNs this behaviour is undesirable.  This PR modifies straylight to only do the NaN-to-zero conversion in a local copy of the input used for the straylight computation itself.

Resolves [JP-3087](https://jira.stsci.edu/browse/JP-3087)

**Checklist for maintainers**
- [x] added entry in `CHANGES.rst` within the relevant release section
- [ ] updated or added relevant tests
- [ ] updated relevant documentation
- [x] added relevant milestone
- [x] added relevant label(s)
- [x] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)
- [ ] Make sure the JIRA ticket is [resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

### CODE REVIEW

**Review by hbushouse — COMMENTED** [review]
Code looks fine. Would it be way too difficult to update the underlying model code to be able to cope with NaNs, as opposed to replacing them before sending down?

File: jwst/straylight/straylight.py [hunk 1]
```diff
@@ -98,18 +98,23 @@ def correct_xartifact(input_model, modelpars):
     # Save some data parameters for easy use later
     nrows, ncols = input_model.data.shape
 
+    # Create a copy of the input data array that will be modified
+    # for use in the straylight calculations
+    input = input_model.copy()
+
     # Create output as a copy of the input science data model
     output = input_model.copy()  # this is used in algorithm to
```
[the remark this answers is not in the document]
Comment (drlaw1558) [reply jwst/straylight/straylight.py:106]: That would require modifying the underlying C code to handle nans gracefully in the computation; it seems safer to clean the inputs ahead of time.

[the remark this answers is not in the document]
Comment (drlaw1558) [reply jwst/straylight/straylight.py]: I don't see why not; updated.

**Review by hbushouse — APPROVED** [review]
LGTM

**Review by jemorrison — APPROVED** [review]
Looks fine to me.

**Comment by hbushouse** [comment]
Regression test run started at https://plwishmaster.stsci.edu:8081/job/RT/job/JWST-Developers-Pull-Requests/573/

**Comment by hbushouse** [comment]
regtest run showed 1 unrelated failure, due to recent changes in stcal/ramp_fitting. So it's good.

**Pull request merged**